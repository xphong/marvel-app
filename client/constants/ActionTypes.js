export const RECEIVE_POWERLEVELS_ERROR = '@@marvel/RECEIVE_POWERLEVELS_ERROR';
export const RECEIVE_POWERLEVELS = '@@marvel/RECEIVE_POWERLEVELS';
export const REQUEST_POWERLEVELS = '@@marvel/REQUEST_POWERLEVELS';

export const RECEIVE_CHARACTERS_ERROR = '@@marvel/RECEIVE_CHARACTERS_ERROR';
export const RECEIVE_CHARACTERS = '@@marvel/RECEIVE_CHARACTERS';
export const REQUEST_CHARACTERS_BY_NAME = '@@marvel/REQUEST_CHARACTERS_BY_NAME';
