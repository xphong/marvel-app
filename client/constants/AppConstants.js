export const CHARACTER_LIMIT = 12;
export const ENDPOINT = '/api/v1';
