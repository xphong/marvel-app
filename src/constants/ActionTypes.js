export const RECEIVE_ERROR = 'RECEIVE_ERROR';
export const RECEIVE_COUNTRIES = 'RECEIVE_COUNTRIES';
export const REQUEST_COUNTRIES = 'REQUEST_COUNTRIES';
