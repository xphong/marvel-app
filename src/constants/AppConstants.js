export const ENDPOINT = 'http://gateway.marvel.com/v1/public/';
export const KEY = '0b72980a074358f7c6565bd7efc480ca';
