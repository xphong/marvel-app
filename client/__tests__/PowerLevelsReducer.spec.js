import * as types from '../constants/ActionTypes';
import powerlevelsReducer from '../reducers/powerLevels';

describe('Power Levels Reducer', () => {
  it('should return initial state', () => {
    const reducer = powerlevelsReducer(undefined, {});

    expect(reducer).toMatchSnapshot();
  });

  it('should handle RECEIVE_POWERLEVELS_ERROR', () => {
    const action = {
      type: types.RECEIVE_POWERLEVELS_ERROR,
      payload: {
        data: []
      }
    };

    const reducer = powerlevelsReducer(undefined, action);

    expect(reducer).toMatchSnapshot();
  });

  it('should handle RECEIVE_POWERLEVELS', () => {
    const action = {
      type: types.RECEIVE_POWERLEVELS,
      payload: {
        data: [{
          name: 'Spider-man'
        }]
      }
    };

    const reducer = powerlevelsReducer(undefined, action);

    expect(reducer).toMatchSnapshot();
  });

  it('should handle REQUEST_POWERLEVELS', () => {
    const action = {
      type: types.REQUEST_POWERLEVELS
    };

    const reducer = powerlevelsReducer(undefined, action);

    expect(reducer).toMatchSnapshot();
  });
});
