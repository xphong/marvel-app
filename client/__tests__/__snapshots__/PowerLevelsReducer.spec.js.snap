// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Power Levels Reducer should handle RECEIVE_POWERLEVELS 1`] = `
Object {
  "data": Array [
    Object {
      "name": "Spider-man",
    },
  ],
  "error": false,
  "isLoading": false,
}
`;

exports[`Power Levels Reducer should handle RECEIVE_POWERLEVELS_ERROR 1`] = `
Object {
  "data": Array [],
  "error": true,
  "isLoading": false,
}
`;

exports[`Power Levels Reducer should handle REQUEST_POWERLEVELS 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isLoading": true,
}
`;

exports[`Power Levels Reducer should return initial state 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isLoading": false,
}
`;
